### ISSUE TITLE
Help updating .icondb.js

### ISSUE THREAD

**Issue by Levertion** [file-icons/atom#634]
I cannot work out how to make .itemdb.js update. I have been trying to update the `Minecraft` entry to include other filetypes, but it won't update, even when running recompile config.
The commit is: 
08b6022d84f93d7c40d261791a10f719f14bd14e

Thanks!

**Comment by Alhadis** [comment]
It should work. Take heed, though, you'll need to reload the editor window before changes will be picked up. I probably should've made that clearer in the hacker's documentation, except there *is* no hacker's documentation. Having said that, I encourage you to share your thoughts on #629. :)

**Comment by Levertion** [comment]
Well I'm using a remote git repo and it isn't picking up any changes. I've saved it multiple times and reloaded + closed and reopened the editor to no avail, in addition to running recompile config again. I'm going to try deleting .icondb (I can always discard that change with git so no danger.)

**Comment by Alhadis** [comment]
Hang on, what do you mean a "remote git repo"? Are the files not located on your own filesystem, on the same partition that Atom's installed (or running) on?

What platform and Atom version are you using?

**Comment by Levertion** [comment]
Wait a moment - I just need to try something. Atom 1.19 windows 10 64 bit.

**Comment by Levertion** [comment]
Oops - didn't mean to close it. Sorry!

**Comment by Alhadis** [comment]
Hey, sorry for the delayed response. Needed to take care of real-world stuff.

Okay, there are two reasons the `compile` utility isn't working for you at the moment:

1. It uses a Unix-style interpreter directive, which means nothing on Windows. A batch-script is needed to accommodate Windows users
2. The `node` executable needs to be reachable from your `%PATH%` variable, which naturally means you'll need [Node](http://nodejs.org/) installed on your system.

That second point brings me to my next question: *is* the Node.js executable installed on your machine?

**Comment by Levertion** [comment]
OK, so the problem is that I don't have node installed on my system? As soon as I get to it I will try and install that. Another potential option is for someone else to compile my fork and gist the output because as soon as I have it compiled I can make a pull and stop trying to compile it. I can probably try by about 1pm BST, but it might not be possible until this evening

**Comment by Alhadis** [comment]
Well... yes, that's one of two problems. The other problem is one that rests with me, which I'm currently working on fixing. I know virtually nothing about Windows batchfile scripts, so some research is also involved. ;p

**Comment by Alhadis** [comment]
Also, please make sure you have the developer dependencies installed by running `apm install .` in the installed package's directory.

**Comment by Levertion** [comment]
OK I will do that as soon as I can.

**Comment by Levertion** [comment]
OK - I have now installed node and run `apm install .` (that said all dependencies were installed successfully) in the file-icons directory, is that all I need to do until you work out how to make it work?

**Comment by Alhadis** [comment]
Patience, please, I'm still trying to work on getting it to update when the file's been changed in Atom. I should stress this part of the package was left half-finished and undocumented on purpose... simply put, I wasn't sure how extensive it needed to be, and I was undecided on a lot of things (which I explained in #629).

But yes, installing `devDependencies` and Node *should* be all you need to do on your end for now. :-)

The `bin/compile` script is currently working from command-line, I'm still trying to figure out why it's not running in response to file changes. ;p You can test it by creating a script at `bin/compile.cmd` with the following source:

~~~batchfile
@IF EXIST "%~dp0\node.exe" (
	"%~dp0\node.exe" "%~dp0\compile" %*
) ELSE (
	@SETLOCAL
	@SET PATHEXT=%PATHEXT:;.JS;=;%
	node -v >nul || (
		echo This script requires NodeJS to run.
		exit /B
	)
	node "%~dp0\compile" %*
)
~~~

**Comment by Levertion** [comment]
Thanks - I didn't mean to sound impatient, sorry about that.
You are much better than many other devs i've met anyway

**Comment by Alhadis** [comment]
\*sigh\* Again, you need to install `devDependencies`. I don't know how this error happened when you said you just finished doing what I asked:

> OK - I have now installed node and run `apm install .` (that said all dependencies were installed successfully) in the file-icons directory

For future reference, any time you see an error that reads `Cannot find module '…'`, 90% of the time it's because you haven't run `apm install .` or `npm install .` (use the former if it's an Atom project, use the latter for ordinary NodeJS projects).

**Comment by Levertion** [comment]
Sorry, I hadn't actually understood what you meant. I was stilling using the command palette version of file-icons compile.
However, when I ran the script in location you specified, it still did not work.
I tried manually installing coffeescript from `https://www.npmjs.com/package/coffee-script` using the command in the package description - however, this installed the package in the `coffeescript` folder rather than a `coffee-script` folder, and there is no reference to `coffee-script` in `package.json` of coffeescript, so I don't think that that is being referenced.

I think I've got it now - I used `npm install coffee-script` and ran the recompile config from the command palette. That finally worked. PR coming soon with my updates.

Thanks for the help!
